flink-streaming-connectors/flink-connector-filesystem/src/test/java/org/apache/flink/streaming/connectors/fs/bucketing/BucketingSinkFaultToleranceITCase.java

```diff
@@ -160,7 +167,15 @@ public void postSubmit() throws Exception {
 				while (line != null) {
 					Matcher matcher = messageRegex.matcher(line);
 					if (matcher.matches()) {
-						numRead++;
+						uniqMessagesRead.add(line);
+
+						// check that in the committed files there are no duplicates
+						if (!file.getPath().toString().endsWith(IN_PROGRESS_SUFFIX) && !file.getPath().toString().endsWith(PENDING_SUFFIX)) {
+							if (!messagesInCommittedFiles.add(line)) {
+								Assert.fail("Duplicate entry in committed bucket.");
```

Comment:
This test will change after introducing a `dispose()` in the `RichFunction`. The reason for the check is that now when we close and given that we do not delete pending/invalid files upon restore, we cannot check the exactly-once so this test checks the "at-least" once. So I would suggest to leave it as is for now and adapt it as soon as the other change gets in. When this happens, the method that was not used will be the new `close()` and it will rename all valid pending files to committed.